src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -329,8 +272,7 @@ public virtual Task InvokeAsync(Func<Task> workItem)
             // This is for example when we run on a system with a single thread, like WebAssembly.
             if (_dispatcher == null)
             {
-                workItem();
-                return Task.CompletedTask;
+                return workItem();
```

Comment:
@javiercn noticed a blazor test fail and this turned out to be the culprit. `workItem()` returns a `Task` and we weren't awaiting it as part of `InvokeAsync`.